language/move-prover/tests/sources/functional/verify_vector.move

```diff
@@ -291,6 +291,23 @@ module 0x42::VerifyVector {
         ensures old(v[i]) == result;
     }
 
+    // Add an element to the vector at index.
+    fun verify_model_insert<Element>(v: &mut vector<Element>, e: Element, i: u64) {
+        vector::insert(v, e, i); // inlining the built-in Boogie procedure
+    }
+    spec verify_model_insert {
```

Comment:
Perhaps, the spec of `verify_remove` can give an idea of how to write the spec for this `insert` function:
```
    spec verify_remove {
        aborts_if j >= len(v);
        ensures len(v) == len(old(v)) - 1;
        ensures v[0..j] == old(v[0..j]);
        ensures v[j..len(v)] == old(v[j+1..len(v)]);
        ensures old(v[j]) == result;
    }
```

Reply:
I was trying to do it, but lacked a bit of expertise in the spec language and missed one trick that would allow a simpler `ensures` clause (fixed now as per previous comment).